go/arrow/csv/writer.go

```diff
@@ -127,3 +127,14 @@ func (w *Writer) Write(record array.Record) error {
 
 	return w.w.WriteAll(recs)
 }
+
+// Flush writes any buffered data to the underlying csv Writer.
+// To check if an error occurred during the Flush, call Error.
+func (w *Writer) Flush() {
```

Comment:
`encoding/csv.Writer.Flush` does indeed expect such a thing, but we could perhaps have this instead:

```go
func (w *Writer) Flush() error {
    w.w.Flush()
    return w.w.Error()
}

func (w *Writer) Error() error { ... }
```

to piggy-back on `bufio.Writer.Flush()` and `os.File.Sync`.

what do you think?